### PR TITLE
Add "ref" input to the prepare-release.yml workflow

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -34,9 +40,16 @@ jobs:
...
       - uses: actions/checkout@v4
+        if: ${{ inputs.checkout-git-ref == false }}
         with:
           token: ${{steps.app-token.outputs.token}}
 
+      - uses: actions/checkout@v4
+        if: ${{ inputs.checkout-git-ref == true }}
+        with:
+          ref: ${{ github.ref }}
+          token: ${{steps.app-token.outputs.token}}
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply .github/workflows/prepare-release.yml:51]: Oh yeah, I could do that. I didn't want to give callers _that_ level of freedom, but if you think it's safe, I'll do it.

[the remark this answers is not in the document]
Comment (gastaldi) [reply .github/workflows/prepare-release.yml:51]: I think it's fine, it should work for other cases as well

[the remark this answers is not in the document]
Comment (Ladicek) [reply .github/workflows/prepare-release.yml:51]: Actually there's a slight difference. `ref: ${{ inputs.ref || github.ref }}` defaults to the `github.ref`, which is the originating branch name, while previous behavior is the exact originating commit. I'm not sure how big of a difference that is, but it felt different enough to add an input to be able to select between the two.

[the remark this answers is not in the document]
Comment (gastaldi) [reply .github/workflows/prepare-release.yml:51]: Dumb question, but you renamed the `checkout-git-ref` to `ref`, right?

[the remark this answers is not in the document]
Comment (gastaldi) [reply .github/workflows/prepare-release.yml:51]: Ah so you need to use the SHA, maybe use `github.head_ref` instead?

[the remark this answers is not in the document]
Comment (Ladicek) [reply .github/workflows/prepare-release.yml:51]: Actually I need to use the branch name :-) It's the current version uses the SHA. I _think_ the default is `''` (which is then replaced by the SHA), but it seems a little weird that would work explicitly. Let me check.

**Review by gastaldi — APPROVED** [review]

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -35,6 +41,7 @@ jobs:
 
       - uses: actions/checkout@v4
         with:
+          ref: ${{ inputs.ref || '' }}
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply .github/workflows/prepare-release.yml:44]: Ah, I see you already fixed that. Thanks!

**Pull request merged**